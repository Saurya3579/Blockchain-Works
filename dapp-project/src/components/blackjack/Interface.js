import React, { Component } from 'react';

import Outcome from './Outcome';

class Interface extends Component {
    static defaultProps = {
        status: "new"
    }

    render() {
        return (
            <div>
                {/* Outcome component child of Interface */}
                <center>
                    <Outcome status={this.props.status} />
                </center>
                <div className="row">
                    <div className="col-lg-6 col-md-6">
                        <h4>Dealer Score : {this.props.dealerscore}</h4>
                    </div>
                    <div className="col-lg-6 col-md-6">
                        <h4 className="text-right">Player Score : {this.props.playerscore}</h4>
                    </div>
                </div>
                <br />
                <div className="row">
                    <div className="col-lg-4 col-md-4">
                        <button onClick={this.props.deal} type="button" className="btn btn-block btn-info">Deal</button>
                    </div>
                    <div className="col-lg-4 col-md-4">
                        {
                            this.props.visible ?
                                this.props.playerscore > 21 || this.props.dealerscore > 21 ?
                                    <button onClick={this.props.hit} type="button" className="btn btn-block btn-success" disabled>
                                        Hit
                                </button>
                                    :
                                    <button onClick={this.props.hit} type="button" className="btn btn-block btn-success">
                                        Hit
                                </button>
                                :
                                <button onClick={this.props.hit} type="button" className="btn btn-block btn-success" disabled>
                                    Hit
                                </button>
                        }
                    </div>
                    <div className="col-lg-4 col-md-4">
                        {
                            this.props.visible ?
                                this.props.playerscore > 21 || this.props.dealerscore > 21 ?
                                    <button onClick={this.props.stand} type="button" className="btn btn-block btn-danger" disabled>
                                        Stand
                                </button>
                                    :
                                    <button onClick={this.props.stand} type="button" className="btn btn-block btn-danger">
                                        Stand
                                </button>
                                :
                                <button onClick={this.props.stand} type="button" className="btn btn-block btn-danger" disabled>
                                    Stand
                                </button>
                        }
                    </div>
                </div>
            </div>
        );
    }
}

export default Interface;
