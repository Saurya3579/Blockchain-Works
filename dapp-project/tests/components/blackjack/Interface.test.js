import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Interface from '../../../src/components/blackjack/Interface';

test('should render NoMatch component', () => {
    const wrapper = shallow(<Interface />);
    expect(toJSON(wrapper)).toMatchSnapshot();
});