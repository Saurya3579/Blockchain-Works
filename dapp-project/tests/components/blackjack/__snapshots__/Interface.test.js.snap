exports[`test should render Interface component 1`] = `
<div>
  <center>
    <Outcome
      status="new" />
  </center>
  <div
    className="row">
    <div
      className="col-lg-6 col-md-6">
      <h4>
        Dealer Score : 
      </h4>
    </div>
    <div
      className="col-lg-6 col-md-6">
      <h4
        className="text-right">
        Player Score : 
      </h4>
    </div>
  </div>
  <br />
  <div
    className="row">
    <div
      className="col-lg-4 col-md-4">
      <button
        className="btn btn-block btn-info"
        type="button">
        Deal
      </button>
    </div>
    <div
      className="col-lg-4 col-md-4">
      <button
        className="btn btn-block btn-success"
        disabled={true}
        type="button">
        Hit
      </button>
    </div>
    <div
      className="col-lg-4 col-md-4">
      <button
        className="btn btn-block btn-danger"
        disabled={true}
        type="button">
        Stand
      </button>
    </div>
  </div>
</div>
`;

exports[`test should render NoMatch component 1`] = `
<div>
  <center>
    <Outcome
      status="new" />
  </center>
  <div
    className="row">
    <div
      className="col-lg-6 col-md-6">
      <h4>
        Dealer Score : 
      </h4>
    </div>
    <div
      className="col-lg-6 col-md-6">
      <h4
        className="text-right">
        Player Score : 
      </h4>
    </div>
  </div>
  <br />
  <div
    className="row">
    <div
      className="col-lg-4 col-md-4">
      <button
        className="btn btn-block btn-info"
        type="button">
        Deal
      </button>
    </div>
    <div
      className="col-lg-4 col-md-4">
      <button
        className="btn btn-block btn-success"
        disabled={true}
        type="button">
        Hit
      </button>
    </div>
    <div
      className="col-lg-4 col-md-4">
      <button
        className="btn btn-block btn-danger"
        disabled={true}
        type="button">
        Stand
      </button>
    </div>
  </div>
</div>
`;
